### PR TITLE
QE: add systems errata API testing feature

### PR DESCRIPTION
**Pull request by NamelessOne91**
## What does this PR change?

Closes https://github.com/SUSE/spacewalk/issues/23534

Adds namespaces and testsuite support for  the system.getRelevantErrata API calls/endpoints and features to test them.

## GUI diff

No difference.

- [x] **DONE**

## Documentation

- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage

- Cucumber tests and files were added and edited

- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/23534
Port(s): Manager 4.3 https://github.com/SUSE/spacewalk/pull/23773

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by jordimassaguerpla** [comment]
Hi ! We fixed a spelling mistake in the name of the tests and you will have to rebase your PR for the correct tests to run. Sorry for the inconvenience.

**Review by Bischoff — APPROVED** [review]

**Review by ktsamis — APPROVED** [review]
Looks good

**Review by nodeg — CHANGES_REQUESTED** [review]
It would make sense to add more labels. Otherwise, the scenarios will fail if the checked minions are not defined and used.

File: testsuite/features/secondary/srv_errata_api.feature [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2024 SUSE LLC
+# Licensed under the terms of the MIT license.
+
+@scope_api
+Feature: Systems errata API
+
+  Scenario: Check the relevant errata for a system
+    When I retrieve the relevant errata for ssh_minion
+
+  Scenario: Check the relevant errata for multiple systems
```
[the remark this answers is not in the document]
Comment (NamelessOne91) [reply testsuite/features/secondary/srv_errata_api.feature:10]: Good point about having these labels, thanks

File: testsuite/features/support/http_client.rb [hunk 2]
```diff
@@ -3,6 +3,11 @@
 
 require 'faraday'
 
+# When we pass a list of values in the query string of an HTTP request, the defaul encoder will only update the key for that param
+# As an example: sids=1000010027&sids=1000010010&sids=1000010012 maps to params={"sids"=>"1000010012"}
+# With FlatParamEncoder, it maps to params={"sids"=>["1000010027", "1000010010", "1000010012"]}
+Faraday::Utils.default_params_encoder = Faraday::FlatParamsEncoder
```
[the remark this answers is not in the document]
Comment (NamelessOne91) [reply testsuite/features/support/http_client.rb:9]: It should not impact the other HTTP requests being done by the testsuite.
It instruct the encoder to not use  the parsing logic  for nested structures, like JSON or XML, which is probably the cause for the `sids` key being updated.
We are explicit on calling `to_json` on the params that are going to form the body a POST request.

Still, it is possible to set the parameters encoder at the individual HTTP request level.
So we can have it for GET requests only and leave the POST ones untouched.

**Review by nodeg — APPROVED** [review]

**Pull request merged**